src/univariate/continuous/kumaraswamy.jl

```diff
@@ -0,0 +1,138 @@
+"""
+    Kumaraswamy(a, b)
+
+The *Kumaraswamy distribution* with shape parameters `a > 0` and `b > 0` has probability
+density function
+
+```math
+f(x; a, b) = a b x^{a - 1} (1 - x^a)^{b - 1}, \\quad 0 < x < 1
+```
+
+It is related to the [Beta distribution](@ref Beta) by the following identity:
+if ``X \\sim \\operatorname{Kumaraswamy}(a, b)`` then ``X^a \\sim \\operatorname{Beta}(1, b)``.
+In particular, if ``X \\sim \\operatorname{Kumaraswamy}(1, 1)`` then
+``X \\sim \\operatorname{Uniform}(0, 1)``.
+
+External links
+
+- [Kumaraswamy distribution on Wikipedia](https://en.wikipedia.org/wiki/Kumaraswamy_distribution)
+
+References
+
+- Kumaraswamy, P. (1980). A generalized probability density function for double-bounded
+  random processes. Journal of Hydrology. 46(1-2), 79-88.
+"""
+struct Kumaraswamy{T<:Real} <: ContinuousUnivariateDistribution
+    a::T
+    b::T
+end
+
+function Kumaraswamy(a::Real, b::Real; check_args::Bool=true)
+    @check_args Kumaraswamy (a, a > zero(a)) (b, b > zero(b))
+    a′, b′ = promote(a, b)
+    return Kumaraswamy{typeof(a′)}(a′, b′)
+end
+
+Kumaraswamy() = Kumaraswamy{Float64}(1.0, 1.0)
+
+Base.convert(::Type{Kumaraswamy{T}}, d::Kumaraswamy) where {T} = Kumaraswamy{T}(T(d.a), T(d.b))
+Base.convert(::Type{Kumaraswamy{T}}, d::Kumaraswamy{T}) where {T} = d
+
+@distr_support Kumaraswamy 0 1
+
+### Parameters
+
+params(d::Kumaraswamy) = (d.a, d.b)
+partype(::Kumaraswamy{T}) where {T} = T
+
+### Evaluation
+
+# `pdf`: Uses fallback `exp(logpdf(_))` method
+
+function logpdf(d::Kumaraswamy, x::Real)
+    a, b = params(d)
+    _x = clamp(x, 0, 1)  # Ensures we can still get a value when outside the support
+    y = log(a) + log(b) + xlogy(a - 1, _x) + xlog1py(b - 1, -_x^a)
+    return insupport(d, x) ? y : (isnan(x) ? oftype(y, NaN) : oftype(y, -Inf))
```

Comment:
Sorry, my previous suggestion was wrong - the following condition should allow to remove the `NaN` special case:
```suggestion
    return x < 0 || x > 1 ? oftype(y, -Inf) : y
```